source/kernel/src/lib.rs

```diff
@@ -1,3 +1,68 @@
+//! # The mnemos kernel
+//!
+//! ## The "kernelspace" entry point
+//!
+//! At the moment, the kernel requires some "entry point" code, more or less the `main()` function
+//! that runs when the system boots, to perform any system-specific initialization.
+//!
+//! This entry point code is responsible for setting up any hardware or subsystems that exist
+//! outside the kernel (or are required by any kernel services), as well as starting and running
+//! the kernel itself.
+//!
+//! ## Creating the kernel
+//!
+//! To create the kernel, you give it a region of memory (as a `*mut u8` + `usize`), by calling
+//! `Kernel::new()`.
+//!
+//! At this point, the system is in "blocking" mode.
+//!
+//! Using the given region of memory, the kernel bootstraps itself, and creates the following:
+//!
+//! * An async executor, intended for "kernel services"
+//! * A kernel service discovery registry
+//! * An async heap allocator, intended for use by kernel services
+//!
+//! After creation, the executor is *not* running yet.
+//!
+//! ## Initialization phase
+//!
+//! After creating the kernel, we need to register kernel services, which are expected to act as
+//! drivers for various system components.
+//!
+//! Since we are not running the executor yet, the kernel provides an interface,
+//! `Kernel::initialize()`, which takes a future and spawns it on the executor. Futures added with
```

Comment:
nit: maybe this should be a link?
```suggestion
//! [`Kernel::initialize()`], which takes a future and spawns it on the executor. Futures added with
```